components/script/dom/webglrenderingcontext.rs

```diff
@@ -991,6 +991,36 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
         Ok(())
     }
 
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
+    fn BufferData_(&self, target: u32, size: i64, usage: u32) -> Fallible<()> {
+        let bound_buffer = match target {
+            constants::ARRAY_BUFFER => self.bound_buffer_array.get(),
+            constants::ELEMENT_ARRAY_BUFFER => self.bound_buffer_element_array.get(),
+            _ => return Ok(self.webgl_error(InvalidEnum)),
+        };
+
+        let bound_buffer = match bound_buffer {
+            Some(bound_buffer) => bound_buffer,
+            None => return Ok(self.webgl_error(InvalidValue)),
+        };
+
+        if size < 0 {
+            return Ok(self.webgl_error(InvalidValue));
+        }
+
+        match usage {
+            constants::STREAM_DRAW |
+            constants::STATIC_DRAW |
+            constants::DYNAMIC_DRAW => (),
+            _ => return Ok(self.webgl_error(InvalidEnum)),
+        }
+
+        let data = vec![0u8; size as usize];
```

Comment:
This is slightly annoying in the sense that we can allocate an unbound buffer when called by the user, but I guess there's no way to allocate this in a fallible way (and there's no need to block on that I guess).

Adding a comment would be helpful if you want :)